app/store/transaction.ts

```diff
@@ -0,0 +1,42 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { VMTransaction } from '../screens/AppNavigator/screens/Transactions/screens/stateProcessor'
+
+type LoadingState = 'idle' | 'loading' | 'loadingMore' | 'success' | 'background' | 'error'
+
+export interface TransactionState {
```

Comment:
Why did you add it on the global state? It's not used anywhere else except Transaction Page

Reply:
My intention is to move the side effects that is not related to the rendering logic outside from the component. 

But I couldn't achieve what I wanted to do when I tried to make async call using redux's createAsyncThunk(). Hence I settled with this solution.

Will there be any scenario or feature in the future that we might need to access transaction data anywhere else? If there is none then I'll move the state out from redux.

Reply:
Nope, we will only use it on this page. Yes please move it out of the global store

Reply:
Updated